torch/_refs/linalg/utils.py

```diff
@@ -0,0 +1,42 @@
+import torch
+from torch._prims.utils import (
+    check,
+    get_higher_dtype,
+    is_half_dtype,
+    is_float_dtype,
+    is_complex_dtype,
+)
+
+from typing import Optional
+
+
+def check_fp_or_complex(dtype: torch.dtype, fn_name: str, half: bool = False):
+    check(
+        is_float_dtype(dtype) or is_complex_dtype(dtype),
+        lambda: f"{fn_name}: Expected a floating point or complex tensor as input. Got {dtype}",
+    )
+    check(
+        half or not is_half_dtype(dtype),
```

Comment:
is_half_dtype can just be `dtype == torch.half`? 

The `half` parameter for this operation might want a more descriptive name like `allow_half`?

Reply:
It can be `float16`, `bfloat16` or `complex32`. I didn't know what's a good name to refer to all these.

Reply:
bfloat16 isn't a half dtype, though -- what about `allow_low_precision_dtypes` or something similar following @ngimel's suggestion?